nixos/qbittorrent: fix legacy default mode on declarative installs (#518616)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2605.section.md

- `services.crabfit` was removed because its upstream packages are unmaintained and insecure.

> - `services.qbittorrent`: when `serverConfig` is non-empty, the module now pre-seeds `BitTorrent/Session/QueueingSystemEnabled = false` to match qBittorrent's documented default. Previously, writing any `serverConfig` caused qBittorrent to misclassify the install as an upgrade from a pre-4.3 version and inject the historical default (`true`), enabling torrent queueing silently. Users who currently have queueing enabled (whether intentionally or through the previous incorrect behavior) must set `serverConfig.BitTorrent.Session.QueueingSystemEnabled = true` explicitly to retain it after upgrading.

Reviewer: This change will not be present in 2605 as it'll be released soon: https://github.com/NixOS/nixpkgs/issues/503391. So we should move this to 2611.

Author: It looks like the 2611 release notes file hasn't been created yet. Do you know what the standard procedure for creating that is?

Reviewer: Ah yes, I didn't think about that. I believe one of the release coordinators will do this. Based on the commit history of the other release notes. So for now we can leave the text in 2605. Once the new release notes are available we could rebase and move.

Author: Done! Rebased onto current master and moved the entry to `rl-2611.section.md` under Backward Incompatibilities.

> - `services.qbittorrent`: when `serverConfig` is non-empty, the module now pre-seeds `BitTorrent/Session/QueueingSystemEnabled = false` to match qBittorrent's documented default. Previously, writing any `serverConfig` caused qBittorrent to misclassify the install as an upgrade from a pre-4.3 version and inject the historical default (`true`), enabling torrent queueing silently. Users who currently have queueing enabled (whether intentionally or through the previous incorrect behavior) must set `serverConfig.BitTorrent.Session.QueueingSystemEnabled = true` explicitly to retain it after upgrading.

Reviewer: ```suggestion
- `services.qbittorrent`: when `serverConfig` is non-empty, the module now pre-seeds `BitTorrent/Session/QueueingSystemEnabled = false` to match qBittorrent's documented default. Previously, writing any `serverConfig` caused qBittorrent to misclassify the install as an upgrade from a pre-4.3 version and inject the historical default (`true`), enabling torrent queueing silently. Users of `serverConfig` that currently haven't set `QueueingSystemEnabled` must set `serverConfig.BitTorrent.Session.QueueingSystemEnabled = true` to retain the historical default after upgrading. Users that don't use `serverConfig` or have already set `QueueingSystemEnabled` are not impacted.
```
User of `serverConfig` that have already set `QueueingSystemEnabled` shouldn't need to change anything right?

Author: Yep, that's definitely clearer wording. Thanks!
